lucet-runtime/lucet-runtime-internals/src/alloc/mod.rs

```diff
@@ -484,10 +488,56 @@ impl Limits {
             heap_memory_size: 16 * 64 * 1024,
             heap_address_space_size: 0x0002_0000_0000,
             stack_size: 128 * 1024,
+            hostcall_reservation: 32 * 1024,
             globals_size: 4096,
             signal_stack_size: DEFAULT_SIGNAL_STACK_SIZE,
+            _hidden: (),
         }
     }
+    pub fn stack_size(&self) -> usize {
+        self.stack_size
+    }
+    pub fn hostcall_reservation(&self) -> usize {
+        self.hostcall_reservation
+    }
+    pub fn with_heap_memory_size(mut self, heap_memory_size: usize) -> Self {
+        self.heap_memory_size = heap_memory_size;
+        self
+    }
+    pub fn with_heap_address_space_size(mut self, heap_address_space_size: usize) -> Self {
+        self.heap_address_space_size = heap_address_space_size;
+        self
+    }
+    pub fn with_stack_size(mut self, stack_size: usize) -> Result<Self, Error> {
+        if stack_size <= self.hostcall_reservation {
+            return Err(Error::InvalidArgument(
+                "stack size must be greater than hostcall reserved space",
+            ));
```

Comment:
As discussed in Slack, we should be able to do these checks in `Limits::validate()` rather than when constructing limits. However, I really like having these interfaces!

Would you mind moving the check, and then removing `_hidden` and making the fields public once more?

As a bonus, with the checks gone these builder methods should all be `const fn`-able.

Reply:
Builder methods are now `const fn`'d, `validate` is absolutely the right place for these checks. Done!